docs/constraints/Scala.md

```diff
@@ -0,0 +1,41 @@
+# Scala Constraints
+
+These are the Scala specific constraints applied to the MetaModel before reaching the presets. [Read here to get a more general idea on the overall process](https://github.com/asyncapi/modelina/blob/master/docs/input-processing.md) of converting a MetaModel to a ConstrainedMetaModel.
+
+## Model Naming
+These are the constraints that is applied to model naming. The `Rule key` is what you can use in the options to overwrite the default behavior. See [constraint customization](https://github.com/asyncapi/modelina/blob/master/docs/README.md#Customization).
+
+|Rule key|Rule|Resolution|
+|---|---|---|
+|NO_SPECIAL_CHAR|No special characters| Special characters are replaced by their name, for example `!` is replaced with `exclamation`. For scala `_` and `$` are an exception to this rule. |
```

Comment:
@officialasishkumar please go through the implementation and make sure it matches the docs: https://github.com/asyncapi/modelina/blob/master/src/generators/scala/constrainer/ModelNameConstrainer.ts

I.e. `'_'` is the exception not `$` 🙂